src/databricks/sql/auth/oauth.py

```diff
@@ -0,0 +1,286 @@
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import base64
+import hashlib
+import json
+import os
+import webbrowser
+
+from datetime import datetime, timedelta, tzinfo
+
+import click
+
+import jwt
+from jwt import PyJWTError
+
+import oauthlib.oauth2
+from oauthlib.oauth2.rfc6749.errors import OAuth2Error
+
+import requests
+from requests.exceptions import RequestException
+
+from databricks_cli.utils import error_and_quit
+
+try:
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+except ImportError:
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+
+
+# This could use 'import secrets' in Python 3
+def token_urlsafe(nbytes=32):
+    tok = os.urandom(nbytes)
+    return base64.urlsafe_b64encode(tok).rstrip(b'=').decode('ascii')
+
+
+# This could be datetime.timezone.utc in Python 3
+class UTCTimeZone(tzinfo):
+    """UTC"""
+    def utcoffset(self, dt):
+        #pylint: disable=unused-argument
+        return timedelta(0)
+
+    def tzname(self, dt):
+        #pylint: disable=unused-argument
+        return "UTC"
+
+    def dst(self, dt):
+        #pylint: disable=unused-argument
+        return timedelta(0)
+
+
+# Some constant values
+OIDC_REDIRECTOR_PATH = "oidc"
+REDIRECT_PORT = 8020
+UTC = UTCTimeZone()
+
+
+def get_client(client_id):
+    return oauthlib.oauth2.WebApplicationClient(client_id)
+
+
+def get_redirect_url(port=REDIRECT_PORT):
+    return "http://localhost:{port}".format(port=port)
+
+
+def fetch_well_known_config(idp_url):
+    known_config_url = "{idp_url}/.well-known/oauth-authorization-server".format(idp_url=idp_url)
+    try:
+        response = requests.request(method="GET", url=known_config_url)
+    except RequestException:
+        error_and_quit("Unable to fetch OAuth configuration from {idp_url}.\n"
+                       "Verify it is a valid workspace URL and that OAuth is "
+                       "enabled on this account.".format(idp_url=idp_url))
+
+    if response.status_code != 200:
+        error_and_quit("Received status {status} OAuth configuration from "
+                       "{idp_url}.\n Verify it is a valid workspace URL and "
+                       "that OAuth is enabled on this account."
+                       .format(status=response.status_code, idp_url=idp_url))
+    try:
+        return json.loads(response.text)
```

Comment:
use `response.json()`, it's a standard method, re-implementing it here only makes it more complicated

Reply:
addressed.